src/frontend/src/tests/lib/services/token.services.spec.ts

```diff
@@ -0,0 +1,177 @@
+import type { SaveUserToken } from '$eth/services/erc20-user-tokens-services';
+import type { Erc20Token } from '$eth/types/erc20';
+import {
+	autoLoadToken,
+	loadTokenAndRun,
+	type AutoLoadTokenParams
+} from '$lib/services/token.services';
+import { busy } from '$lib/stores/busy.store';
+import * as toastsStore from '$lib/stores/toasts.store';
+import { token } from '$lib/stores/token.store';
+import { mockIdentity } from '$tests/mocks/identity.mock';
+import { mockValidToken } from '$tests/mocks/tokens.mock';
+import { get } from 'svelte/store';
+import type { Mock, MockInstance } from 'vitest';
+
+describe('token.services', () => {
+	const mockCallback = vi.fn();
+
+	beforeEach(() => {
+		vi.clearAllMocks();
+	});
+
+	describe('loadTokenAndRun', () => {
+		it('should set the token in tokenStore', async () => {
+			await loadTokenAndRun({ token: mockValidToken, callback: mockCallback });
+			const tokenStore = get(token);
+			expect(tokenStore).toBe(mockValidToken);
+		});
+
+		it('should call the callback function', async () => {
+			await loadTokenAndRun({ token: mockValidToken, callback: mockCallback });
+
+			expect(mockCallback).toHaveBeenCalled();
+		});
+
+		it('should handle asynchronous callback', () =>
+			new Promise<void>((done) => {
+				// eslint-disable-next-line require-await
+				const callback = async () => {
+					done();
+				};
+
+				loadTokenAndRun({ token: mockValidToken, callback });
```

Comment:
i am not really sure here, how is this test "tested"?

Reply:
With `done`. If it is not called, the test fails. It assert the promise of the callback is correctly handled.

Reply:
ah! right!!! thank you for the explanation